readme-developer.md

```diff
@@ -0,0 +1,158 @@
+
+- Go to Settings > Compiler > Global Compiler Settings > Toolchain Executables. Set the **Compiler's installation directory** to `C:\Program Files\mingw-w64\x86_64-8.1.0-posix-seh-rt_v6-rev0\mingw64\bin`, if you followed the default MinGW installation process above.
+- Modify **Linker for static libs** from `ar.exe` to `gcc-ar.exe`.
+
+After this, the game should be ready to compile. Use CodeBlocks to open EndlessSky.workspace in the directory where you've cloned the game's code. This workspace consists of three projects: one for the majority of source code, one for the game binary, and one for the unit tests. As with most IDE-based projects, you must explicitly add new files to the respective project for them to be compiled.
+
+With the workspace open, you can double-click a project in the left-hand menu to activate that project, binding the toolbar & keyboard shortcuts for "Build," "Run,", "Build & Run," etc. to it, no matter which file is being edited.
+
+For example, when making lots of changes to the game, you will generally have the "endless-sky-lib" project active, so that you can ensure changes compile without needing to fully link things together into the actual game binary. After making changes, you would then activate the "EndlessSky" project and use the "Build & Run" option with the "Release" target, which will fully link the compiled code into the small, performant executable.
+
+You can continue making tweaks to files that belong to the "endless-sky-lib" project without reactivating it, so that "Build & Run" will still launch the game with your latest tweaks. Should you experience some show-stopper bug in your modified game, you would then change to the "Debug" build target to help determine the source(s) of the issue.
+
+Before the game will run the first time, however, you will need to skip down to the instructions at **Running the game** below.
+
+### Building with Scons
+
+If you instead want to build the game from the command line via Scons, you will need [Python 3.8 or later](https://www.python.org/downloads/). When installing, be sure to select the "Add to PATH" checkbox.
+
+Afterwards, you will need to add your MinGW installation to your path manually. To do so on Windows 10, to to **Settings** > **System** > **About** > **System info** > **Advanced system settings** > **Environment Variables**. From each, select **Path** under System variables, and click Edit.
+
+Click New, and then if you used a default MinGW installation, enter `C:\Program Files\mingw-w64\x86_64-8.1.0-posix-seh-rt_v6-rev0\mingw64\bin` as the path, then hit OK to exit out of these windows.
+
+From there, restart your command line if you had one open, and simply `cd` to your Endless Sky directory, then type `mingw32-make.exe -f .winmake`. To speed up the build process, you can add a `-jX` flag to that command, where X is the number of CORES/THREADS? your processor has.
```

Comment:
```suggestion
From there, restart your command line if you had one open, and simply `cd` to your Endless Sky directory, then type `mingw32-make.exe -f .winmake`. (`make`'s standard `-j N` flag is supported, to increase the number of active parallel build tasks to N.)
```

Reply:
What should a user put in for N? Is it based on threads or cores? I'm guessing threads, but just want to be sure.

Reply:
Neither, as with all parallelism the choice of "how many jobs" is dependent on both the system and the amount of work to be done.

Reply:
Can we offer a guideline? This is meant for new developers, and the workload is relatively fixed. It doesn't take more than 5 minutes on my terrible laptop. So using a quad-core processor as an example, like an i5-6600k, how many jobs would you run?

Reply:
the most common use case is maxing out your CPU, which is achieved by N = Number of logical cores. You may need to clarify this use case to make tehh happy.

Reply:
We shouldnt need to provide any guideline on the value. Explaining how to invoke parallel builds should be enough to get developers started, either learning about "make" and the `-j` flag (because they're new to compiling in general) or just to building (e.g. because they're new to `scons` and didn't know it supports parallel builds, too), or experimenting with different values (because they're curious about the possible speedups).

As the intended audience of this document is people familiar with building software programs, or at minimum intent on learning to build this program, I think it is not our place to explain how other programs work - those programs have entire websites devoted to them with that purpose!

Reply:
We don't _need_ to, sure, but the whole point of a guide is to provide guidance. If MCO is right, and the most common use case is to put in your CPU's logical processors for the number of tasks, then I see no harm mentioning that. If not, and you really need to do lots of in depth reading or experimentation to find anything close to useful, then it's fine as-is.

Reply:
As a new developer for whom scons is a misspelling of a tasty bread product rather than something to do with games, having the guide spell out common flags and how to use them would be appreciated. It is a guide, after all, and having it be all-inclusive as to what it takes for me to build things quickly and effectively would be much appreciated.

The main thing, I suppose, is to not make any assumptions about our developer's knowledge.